go/pkg/websocket/websocket.go

```diff
@@ -133,11 +135,17 @@ func (c *Connection) Stop() {
 func (c *Connection) cleanup() {
 	<-c.doneChan
 	c.ticker.Stop()
-	_ = c.conn.WriteMessage(websocket.CloseMessage, []byte{})
+	_ = c.send(websocket.CloseMessage, []byte{})
 	c.conn.Close()
 	close(c.msgChan)
 }
 
+func (c *Connection) send(messageType int, data []byte) error {
+	c.m.Lock()
+	defer c.m.Unlock()
```

Comment:
not a golang wizard but i assume this defer unlock will unlock the mutex when we exit the scope of this fn?

Reply:
correct